src/outputs/parthenon_hdf5.hpp

```diff
@@ -255,6 +272,72 @@ std::vector<T> HDF5ReadAttributeVec(hid_t location, const std::string &name) {
 template <>
 std::vector<std::string> HDF5ReadAttributeVec(hid_t location, const std::string &name);
 
+template <>
+std::vector<bool> HDF5ReadAttributeVec(hid_t location, const std::string &name);
+
+void HDF5ReadAttribute(hid_t location, const std::string &name, std::string &val);
+
+template <typename T, REQUIRES(implements<scalar(T)>::value)>
+void HDF5ReadAttribute(hid_t location, const std::string &name, T &val) {
+  auto vec = HDF5ReadAttributeVec<T>(location, name);
+  val = vec[0];
+}
+
+template <typename T, REQUIRES(implements<kokkos_view(T)>::value)>
+void HDF5ReadAttribute(hid_t location, const std::string &name, T &view) {
+  static_assert(std::is_same<typename T::array_layout, Kokkos::LayoutLeft>::value ||
+                    std::is_same<typename T::array_layout, Kokkos::LayoutRight>::value,
+                "Currently can only read from contiguous views");
+  // attribute info
+  H5A attr;
+  auto [rank, dim, size] = HDF5GetAttributeInfo(location, name, attr);
+
+  // check rank
+  int view_rank = static_cast<size_t>(T::rank);
+  PARTHENON_REQUIRE(rank == view_rank, "input and output view are same rank");
+
+  // Resize view.
+  typename T::array_layout layout;
+  for (int d = 0; d < rank; ++d) {
+    layout.dimension[d] = dim[d];
+  }
+  view = T(view.label(), layout);
+
+  // pull out data pointer
+  auto *pdata = view.data();
+  auto view_h = Kokkos::create_mirror_view(view);
+  if constexpr (!std::is_same<typename T::memory_space, Kokkos::HostSpace>::value) {
+    Kokkos::deep_copy(view_h, view);
+    pdata = view_h.data();
+  }
+
+  // check type
+  auto type = getHDF5Type(pdata);
+  const H5T hdf5_type = H5T::FromHIDCheck(H5Aget_type(attr));
+  auto status = PARTHENON_HDF5_CHECK(H5Tequal(type, hdf5_type));
+  PARTHENON_REQUIRE_THROWS(status > 0, "Type mismatch for attribute " + name);
+
+  // Read attribute from file
+  PARTHENON_HDF5_CHECK(H5Aread(attr, type, pdata));
+
+  if constexpr (!std::is_same<typename T::memory_space, Kokkos::HostSpace>::value) {
+    Kokkos::deep_copy(view, view_h);
+  }
+}
+
+template <typename D, typename S>
+void HDF5ReadAttribute(hid_t location, const std::string &name,
+                       ParArrayGeneric<D, S> &pararray) {
+  // forces compiler to pass by non-const reference into the next function
+  D &view = pararray.KokkosView();
+  HDF5ReadAttribute(location, name, view);
+}
```

Comment:
Obviously this means that anything stored in the `ParArray` state is lost when restarting. I think this is fine (since the state is more to easily transfer properties of fields between host and device), but we may want to add a comment somewhere to that effect.